Feature/list documents

ivancolomer opened the pull request:
Added Create, Edit and List component for documents

You can also upload files into document entity

ivancolomer committed:
Added document-list with document and documentService

ivancolomer committed:
Added route for document-list

ivancolomer committed:
Added find documents by selectionProcess

ivancolomer committed:
Merge branch 'main' of https://github.com/UdL-EPS-SoftArch/AcademicRecruit-GEIADE into feature/list-documents

# Conflicts:
#	package-lock.json
#	src/app/app-routing.module.ts
#	src/app/app.module.ts

ivancolomer committed:
Merge branch 'main' of https://github.com/UdL-EPS-SoftArch/AcademicRecruit-GEIADE into feature/list-documents

# Conflicts:
#	package-lock.json
#	src/app/app-routing.module.ts
#	src/app/app.module.ts

ivancolomer committed:
Added document page list for a selection process

ivancolomer committed:
Fixed document list component, added file service, added angular material dependencies

ivancolomer committed:
Added create component for documents

ivancolomer committed:
Added edit component for documents
It lets you upload files and shows a progress spinner while uploading

rogargon commented:
The added functionality is not yet linked from the navigation bar.
Regarding the newly introduced components from Material, are they strictly necessary? Mixing different layout frameworks might introduce additional complexity into the application and increase significantly its size

ivancolomer committed:
1. Removed @angular Material for this project.
2. Added bootstrap modal with progress bar for the file upload.
3. Added buttons for download and upload.
4. Added open/read functionality for the file.
5. Added delete file in a document.
6. Added buttons download and open on document list.
7. Added delete document on document list.

ivancolomer committed:
Merge branch 'main' of https://github.com/UdL-EPS-SoftArch/AcademicRecruit-GEIADE into feature/list-documents

# Conflicts:
#	src/app/app-routing.module.ts
#	src/app/app.module.ts
#	src/app/login-basic/user.spec.ts
#	src/app/selection-process/selection-process.ts

ivancolomer commented:
We are wating for selection process list and edit pull request to finish so we can merge documents list component into selection process page; therefore, there won't be a link in navbar for documents (as it happens for users and selection processes)

ivancolomer committed:
Removed material styles on styles.css

rogargon committed:
Merge branch 'main' into feature/list-documents
